lib/taskana-core-test/src/test/java/acceptance/classification/delete/DeleteClassificationAccTest.java

```diff
@@ -0,0 +1,240 @@
+        TaskAttachmentBuilder.newAttachment()
+            .classificationSummary(attachmentClassification.asSummary())
+            .objectReference(DefaultTestEntities.defaultTestObjectReference().build())
+            .build();
+    TaskBuilder.newTask()
+        .classificationSummary(classification.asSummary())
+        .workbasketSummary(defaultWorkbasket)
+        .attachments(attachment)
+        .primaryObjRef(DefaultTestEntities.defaultTestObjectReference().build())
+        .buildAndStore(taskService);
+
+    assertThatThrownBy(
+            () -> classificationService.deleteClassification(attachmentClassification.getId()))
+        .isInstanceOf(ClassificationInUseException.class);
+  }
+
+  @WithAccessId(user = "businessadmin")
+  @Test
+  void should_ThrowExceptionAndRollback_When_ChildClassificationIsInUse() throws Exception {
+    Classification parent =
+        DefaultTestEntities.defaultTestClassification().buildAndStore(classificationService);
+    Classification child =
+        DefaultTestEntities.defaultTestClassification()
+            .parentId(parent.getId())
+            .buildAndStore(classificationService);
+    TaskBuilder.newTask()
+        .primaryObjRef(DefaultTestEntities.defaultTestObjectReference().build())
+        .classificationSummary(child)
+        .workbasketSummary(defaultWorkbasket)
+        .buildAndStore(taskService);
+
+    assertThatThrownBy(() -> classificationService.deleteClassification(parent.getId()))
+        .isInstanceOf(ClassificationInUseException.class);
+    Classification rollback =
+        classificationService.getClassification(parent.getKey(), parent.getDomain());
+    assertThat(rollback.getDomain()).isEqualTo(parent.getDomain());
+  }
+
+  @WithAccessId(user = "businessadmin")
+  @Test
+  void should_ThrowExceptionAndRollbackAllDomains_When_ChildClassificationIsInUse()
+      throws Exception {
+    Classification parent =
+        DefaultTestEntities.defaultTestClassification().buildAndStore(classificationService);
+    Classification child =
+        DefaultTestEntities.defaultTestClassification()
+            .parentId(parent.getId())
+            .buildAndStore(classificationService);
+    TaskBuilder.newTask()
+        .primaryObjRef(DefaultTestEntities.defaultTestObjectReference().build())
+        .classificationSummary(child)
+        .workbasketSummary(defaultWorkbasket)
+        .buildAndStore(taskService);
+
+    assertThatThrownBy(() -> classificationService.deleteClassification(parent.getKey(), ""))
+        .isInstanceOf(ClassificationInUseException.class);
+    Classification rollbackMaster = classificationService.getClassification(parent.getKey(), "");
+    Classification rollbackDomain =
+        classificationService.getClassification(parent.getKey(), parent.getDomain());
+    assertThat(rollbackDomain.getKey()).isEqualTo(rollbackMaster.getKey());
+    assertThat(rollbackDomain.getDomain()).isNotEqualTo(rollbackMaster.getDomain());
+  }
```

Comment:
I would propose the same assertion method as mentioned for `should_ThrowExceptionAndRollback_When_ChildClassificationIsInUse()`. Further, i would do the same assertion method for rollbackMaster by using a new variable expectedMasterClassification and assert equality with masterClassification.

Reply:
Using the method of assertion is a great idea with rollbackDomain! I cannot create an expectedMasterClassification using the api because the id of each created classification changes.

Reply:
You could copy the id of the master classification or ignore the id field with  assertThat(rollbackMaster).isEqualTo(expectedMaster).usingRecursiveComparison().ignoringFields("id").

Reply:
Sounds good :)